### PR TITLE
FIX: prevent anonymous users from changing their email/username/name

### PR DESCRIPTION
**Pull request by majakomel**
https://meta.discourse.org/t/suspect-account-anonymous-account/112813

### CODE REVIEW

File: spec/components/guardian_spec.rb [hunk 1]
```diff
@@ -2403,6 +2404,20 @@
       it "is true for admins" do
         expect(Guardian.new(admin).can_edit_username?(user)).to be_truthy
       end
+
+      it "is true for admins when changing anonymous username" do
+        expect(Guardian.new(admin).can_edit_username?(anonymous_user)).to be_truthy
```
Comment (tgxworld) [inline spec/components/guardian_spec.rb]: Should admin be able to edit usernames for anon users as well?

File: test/javascripts/controllers/preferences-account-test.js.es6 [hunk 2]
```diff
@@ -21,6 +22,10 @@ QUnit.skip("updating of associated accounts", function(assert) {
 
   assert.equal(controller.get("canUpdateAssociatedAccounts"), false);
 
+  controller.set("model.is_anonymous", false);
+
+  assert.equal(controller.get("canUpdateAssociatedAccounts"), false);
```
Comment (tgxworld) [inline test/javascripts/controllers/preferences-account-test.js.es6]: If the mode is not anonymous, should we be able to update the associated accounts?

Comment (majakomel) [reply test/javascripts/controllers/preferences-account-test.js.es6]: Non-anonymous users are able to update associated accounts if other conditions are met https://github.com/discourse/discourse/pull/7311/files#diff-4aba2c210bc3e273a9f57c8348785fd4R109

**Comment by majakomel** [comment]
@SamSaffron, should admin still be able to edit email/username/name of anonymous users after the change?

I removed password and associated account from preferences and was thinking that it would make sense to disable editing of about/location/website/avatar as well to prevent anonymous account from being too personalized?

File: test/javascripts/controllers/preferences-account-test.js.es6 (hunk 2 above)
Comment (tgxworld) [reply test/javascripts/controllers/preferences-account-test.js.es6]: ah icic. There are other conditions 👍

**Comment by ZogStriP** [comment]
I agree we should prevent any kind of customization for Anonymous users but still allow Admins to change anything they want.

@SamSaffron what do you think?

**Comment by SamSaffron** [comment]
I think admins are fine to make changes here if they wish. I am merging this.

**Pull request merged**